zerver/lib/import_realm.py

```diff
@@ -27,6 +28,10 @@
     CustomProfileField, CustomProfileFieldValue, \
     Attachment, get_system_bot, email_to_username
 
+
+MESSAGE_URL_REGEX = ("(%s[a-z0-9-:.]+)/(#narrow/stream/([^/]+)/subject/[^/]+/near/([0-9]+))" %
+                     settings.EXTERNAL_URI_SCHEME)
```

Comment:
Hmm.  This may not be generic enough for the general case; e.g. PM threads will have a slightly different format....

But I think it could handle the Slack import case, so might be good enough for now?